numpy/f2py/crackfortran.py

```diff
@@ -2873,6 +2906,84 @@ def analyzevars(block):
 analyzeargs_re_1 = re.compile(r'\A[a-z]+[\w$]*\Z', re.I)
 
 
+def param_eval(n, varsn, v, g_params, params):
+    # Creates a dictionary of indices and values for each parameter in a
+    # parameter array to be evaluated later.
+    #
+    # WARNING: It is not possible to initialize multidimensional array
+    # parameters e.g. dimension(-3:1, 4, 3:5) at this point. This is because in
+    # Fortran initialization through array constructor requires the RESHAPE
+    # intrinsic function. Since the right-hand side of the parameter declaration
+    # is not executed in f2py, but rather at the compiled c/fortran extension, later, it is not possible
+    # to execute a reshape of a parameter array.
+    # One issue remains: if the user wants to access the array parameter from python,
+    # we should either
+    # 1) allow them to access the parameter array using python standard indexing (which
+    #    is often incompatible with the original fortran indexing)
+    # 2) allow the parameter array to be accessed in python as a dictionary with fortran
+    #    indices as keys
+    # We are choosing 2 for now.
```

Comment:
One option here would be:
```python
class FortranParamArray:
    def __init__(self, indices, values):
        self._indices = indices
        self._values = values
    def __getitem__(self, index):
        return self._values[self._indices.index(index)]
```

Reply:
And pass the `range` object in as the `indices` argument.

Reply:
I didn't consider this because the f2py code does not use OOP and I think this might be a part of a larger refactor in the future.